pkg/reconcile/pipeline/infinispan/handler/provision/statefulsets.go

```diff
@@ -85,6 +85,11 @@ func ClusterStatefulSetSpec(statefulSetName string, i *ispnv1.Infinispan, ctx pi
 	statefulSetAnnotations["checksum/podEnvs"] = podEnvsHash
 	annotationsForPod := i.PodAnnotations()
 	annotationsForPod["updateDate"] = time.Now().String()
+	gracePeriod := int64(30) // default fallback
+
+	if i.Spec.Service.Container.TerminationGracePeriodSeconds != nil {
```

Comment:
`i.Spec.Service.Container` is a pointer, so it's possible that this value is `nil`. We need to guard against this to prevent Nil Pointer Deference errors being thrown.

I suggest to add a helper method to `types_util.go` similar to `StorageClassName` that returns `nil` if `spec.Service.Container` is nil otherwise it just returns the value of `TerminationGracePeriodSeconds` (can be nil).